clients/command/src/main/java/org/eclipse/hono/client/command/CommandRouterCommandConsumerFactory.java

```diff
@@ -96,10 +102,25 @@ private void reenableCommandRouting() {
     }
 
     private static String getNewAdapterInstanceId(final String adapterName) {
+        final String k8sContainerId = ContainerUtil.getContainerId();
```

Comment:
what do we need `ContainerUtil` for? Isn't it sufficient to get the pod ID via environment variables?

Reply:
[ContainerUtil](https://github.com/apache/logging-log4j2/blob/5edf43758f104affad9842753e66be173fc4d68a/log4j-kubernetes/src/main/java/org/apache/logging/log4j/kubernetes/ContainerUtil.java) is used for identifying the docker container ID of the container that this code is running in.

As argued in https://github.com/eclipse/hono/issues/2028#issuecomment-875608809, the pod name/id alone isn't suitable for the adapter instance id here, therefore the container ID needs to be identified. Unfortunately, there is currently no way of exposing the container id in an environment variable via the Kubernetes downward API, as it is possible with the pod name for example. Therefore, `/proc/self/cgroup` is used via the `ContainerUtil` class.

Reply:
Ok, I see